arduino/WandCore/WandCore.ino

```diff
@@ -253,72 +257,114 @@ void loop()
-        incrementRGBColorloop(); 
-        debugI("Switched to Color Mode");
-      } else {
-        wandMode = POWER;
-        setRGBLED(255, 255, 255); // white for power mode
-        debugI("Switched to Power Mode"); 
-      }      
-      
-      break;
-    case HOLDING:
-      // TODO adjust brightness or color depending on mode
-      if ((millis() - lastLogged) >= 500) { // DEBUG ONLY, throttling log output
-        debugI("touch is being held");
-        lastLogged = millis();
-      }
+  if (currentTouchEvent == SINGLE_TAP) {
+    debugI("Single Tap Just Happened");
+    if (wandMode == POWER) {                      // power mode, single tap == set light brightness based on "brightness" of color sensed
+      // TODO
+    }
+    else if (wandMode == COLOR) {                 // color mode, single tap == set color of light based on color sensed
+      // TODO
+    }
       
-      break;
+  } else if (currentTouchEvent == DOUBLE_TAP) {
+    debugI("Double Tap Just Happened");
+    // toggle the wand's mode
+    if (wandMode == POWER) {                      // power mode, double tap == switch to color mode
+      wandMode = COLOR;
+      incrementRGBColorloop();                    // resume cycling the onboard RGB through its colorloop
+      debugI("Switched to Color Mode");
+    } else {                                      // color mode, double tap == switch to power mode
+      wandMode = POWER;
+      setRGBLED(255, 255, 255);                   // solid white for power mode
+      debugI("Switched to Power Mode"); 
+    }
   }
-  
-  // Read IMU sensor
-  readIMU(imuData);
 
-  // Calculate sum of acceleration and compare to threshold
+  // Calculate sum of acceleration to compare to threshold
   float aSum = fabs(imuData[AX]) + fabs(imuData[AY]) + fabs(imuData[AZ]);
   float gSum = fabs(imuData[GX]) + fabs(imuData[GY]) + fabs(imuData[GZ]);
 
-  // Acceleration threshold reached, predict gesture
+  // Acceleration threshold reached, handle gesture
   if (aSum >= accelerationThresholdG || gSum >= gyroThreshold) {
+    // do the gesture predicting
     readAndPredictGesture(imuData, gestureConfidence);
     
-    if (gestureConfidence[GESTURE_FLICK] > 0.9) { // it's a flick, do flick things!
+    // read target light state
+    // only do this here as it's not needed elsewhere and this minimizes polling
+    // step 1 of 3 - prepare the raw string from the Hue library
+    String rawLightState;
+    rawLightState = myHue.getLightInfo(lightID);
+    int removeToIdx = rawLightState.indexOf("{") - 1;
+    rawLightState.remove(0, removeToIdx);
+
+    // step 2 of 3 - parse the JSON
+    StaticJsonDocument<112> filter;
+    JsonObject filter_state = filter.createNestedObject("state");
+    filter_state["hue"] = true;
+    filter_state["on"] = true;
+    filter_state["effect"] = true;
+    filter_state["bri"] = true;
+    filter_state["sat"] = true;
+    filter_state["ct"] = true;
+    
+    StaticJsonDocument<192> doc;

```

Comment:
Can you comment on the magic numbers `112` and `192` here?

Reply:
will add a comment in the code, but they come from using the ArduinoJson assistant https://arduinojson.org/v6/assistant/ that recommends the size of these documents based on examples you feed it. Generated nifty custom sample code too!